### DISCUSSION TITLE
Head component not working

### DISCUSSION THREAD

**Discussion by detronetdip** [vercel/next.js#53529] [Help] [answered]
### Summary

I am using the `Head` component from `next/head` in my project to set the title of a page like below.
```tsx
export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <Head>
        <title key="title">vg</title>
      </Head>
      <body className={inter.className}>{children}</body>
    </html>
  );
}
```

but it is not working, when i open the browser the title is not there, I also inspect and saw that there is no title tag in the head tag in html. 

Can anyone please tell me how to make things work?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The next/head component does not work in App Router. You can use the `metadata` object instead. 

- https://nextjs.org/docs/app/building-your-application/optimizing/metadata#static-metadata

Head is only listed for `Pages Router`:

- https://nextjs.org/docs/pages/api-reference/components/head

It is nowhere to be found in App Router docs:

<img width="1728" alt="Screenshot 2023-08-03 at 16 58 01" src="https://github.com/vercel/next.js/assets/21013447/aace640f-32d4-4912-992f-cc430ab5e64e">

**Reply by detronetdip** [reply]
Thanks @icyJoseph

**Reply by adamsh231** [reply]
thank u sir @icyJoseph

**Reply by kitsunekyo** [reply]
how would I add splash screens for my pwa then?

```tsx
<link
  rel="apple-touch-startup-image"
  media="screen and (device-width: 430px) and (device-height: 932px) and (-webkit-device-pixel-ratio: 3) and (orientation: portrait)"
  href="/pwa/splash_screens/iPhone_15_Pro_Max__iPhone_15_Plus__iPhone_14_Pro_Max_portrait.png"
/>
```

**Reply by hskang9** [reply]
Hey, I am trying to use metadata in a page while using useState. Using Head component made me to get the data from the page in client and change it. However, using metadata only works when a page is server side rendered. Is there any workaround in this? Current migration experience is just awful.

**Reply by kodeknight1** [reply]
Ho can I add a custom link in the head? Like a manifest.json link for a PWA?
`<link rel="manifest" href="manifest.json" />`
Such a trivial and common thing to do, yet there seems to be no way of doing it in Nextjs.

**Reply by kitsunekyo** [reply]
`metadata.appleWebApp.startupImage` was the answer for me.

https://www.mostviertel.tech/blog/nextjs-pwa

**Reply by kitsunekyo** [reply]
@hskang9 you might want to check out `generateMetadata()`

https://nextjs.org/docs/app/api-reference/functions/generate-metadata

**Reply by aliamiri1380** [reply]
> Hi,
> 
> The next/head component does not work in App Router. You can use the `metadata` object instead.
> 
> * https://nextjs.org/docs/app/building-your-application/optimizing/metadata#static-metadata
> 
> Head is only listed for `Pages Router`:
> 
> * https://nextjs.org/docs/pages/api-reference/components/head
> 
> It is nowhere to be found in App Router docs:


so my app has dynamic content and data is fetched from api and metadata are adjusted based on api result, so since we have two separate function for metadata and page content it means that to render both metatags and page content i should fetch api twice ???
if yes so its very suboptimal ...

**Reply by crispiestsquid** [reply]
This answer doesn't support the inclusion of `<style>` or `<link rel="stylesheet"` tags. This is a pretty big deal for a project I am working on right now.

Resorting to client side tag injection into the head seems silly and not very performant.

Having to consider running `pages` and `app` router side by side for now until this is supported

**Comment by evgenyfadeev** [comment]
I'm trying to insert `<meta property="fb:app_id" content="..." />` the metadata or generateMetaData way is broken as it does not accept the attributes other than name. So there is no way to do this in Nextjs 14?

**Reply by DanielOberlechner** [reply]
Found a way to do this:
![Screenshot 2024-10-31 at 17 24 01](https://github.com/user-attachments/assets/bc2b95a3-21fd-4768-b2d9-fbb2cdc66403)

Just use metadata:

import type {Metadata, Viewport} from "next";
export const metadata: Metadata = {}

I'm using it in my layout.tsx file :)

**Comment by rollthecoder** [comment]
How can I add custom <style> tag in app router?

**Reply by DanielOberlechner** [reply]
I have tried to create a head tag in my layout.tsx file and add there some custom properties. In theory it should also work wit style tags but I'm not sure to 100%. You could give it a try ...
